src/documents/CoreAPI.md

```diff
@@ -345,6 +367,29 @@ var CATTyears = fluid.hashToArray(hash, "species", function (newElement, oldElem
 Converts an array consisting of a mixture of arrays and non-arrays into the concatenation of any inner arrays 
 with the non-array elements. The original array *** will not be modified ***. See description of [mapcat](http://martinfowler.com/articles/collection-pipeline/flat-map.html) or `flat-map`.
 
+### fluid.freezeRecursive(tofreeze)
+
+* `tofreeze {Array|Object}` The object to be frozen recursively
+* Returns: `{Array|Object}` The supplied argument, recursively frozen
+
+Calls [`Object.freeze`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) at each level of containment of the supplied object.
+This function is a no-op if a primitive value is supplied.
+
+### fluid.pushArray(holder, member, topush)
```

Comment:
I was playing with this interesting function in a debugger console for my little curiosity, I noticed, if supplied `member` value points to an existing object path, and the value of that leaf is not an array, the error `TypeError: array.push is not a function` occurs. If the leaf value is an array, `topush` will be pushed to the end of the array.

The test snippet that throws error:
```javascript
var holder = {"a": "1"};
fluid.pushArray(holder, "a", 3); // holder.array now holds [3]

console.log(holder);
```

Is this the use case where this function should not be used? Or the function should actually turn the leaf value into an array, that means the above example should output `{"a": ["1", 3]}`?